Assorted `run-make-support` maintenance

jieyouxu opened the pull request:
This PR should contain no functional changes.

- Commit 1: Removes the support library's CHANGELOG. In the very beginning, I thought maybe we would try to version this library. But this is a purely internal test support library, and it's just extra busywork trying to maintain changelog/versions. It's also hopelessly outdated.
- Commit 2: Resets version number to `0.0.0`. Ditto on busywork.
- Commit 3: Bump `run-make-support` to Edition 2024. The support library was already "compliant" with Edition 2024.
- Commit 4: Slightly organizes the support library dependencies.
- Commit 5: Previously, I tried hopelessly to maintain some manual formatting, but that was annoying because it required skipping rustfmt (so export ordering etc. could not be extra formatted). Give up, and do some rearrangements / module prefix tricks to get the `lib.rs` looking at least *reasonable*. IMO this is not a strict improvement, but I rather regain the ability to auto-format it with rustfmt.
- Commit {6,7}: Noticed in rust-lang/rust#143669 that we apparently had *both* {`is_msvc`, `is_windows_msvc`}. This PR removes `is_msvc` in favor of `is_windows_msvc` to make it unambiguous (and only retain one way of gating) as there are some UEFI targets which are MSVC but not Windows.

Best reviewed commit-by-commit.

r? @Kobzol

jieyouxu committed:
Remove `run-make-support` CHANGELOG

Hopelessly outdated, and this support library is purely internal.

jieyouxu committed:
Don't attempt to version `run-make-support`

Purely internal test support library.

jieyouxu committed:
Bump `run-make-support` to Edition 2024

jieyouxu committed:
Sort and document `run-make-support` dependencies

jieyouxu committed:
Massage `lib.rs` so it can be rustfmt'd

Even if the formatting isn't strictly "better", it at least allows
rustfmtting it automatically.

jieyouxu committed:
Use `is_windows_msvc` instead of `is_msvc` in run-make tests

jieyouxu committed:
Only provide `is_windows_msvc` to gate on windows-msvc

And not both `is_windows_msvc` and `is_msvc`.

Kobzol commented:
:+1: on removing the changelog, I also thought several times that it's futile to try to do this. Also looks like all targets containing `msvc` currently end with `windows-msvc`, so it shouldn't be a behavior change.

You can r=me after CI is green.

jieyouxu commented:
> Also looks like all targets containing `msvc` currently end with `windows-msvc`, so it shouldn't be a behavior change.

I would be very concerned if we had non-Windows targets that ended with `msvc` 😆

petrochenkov commented:
UEFI targets are MSVC, but not windows (in terms of `rustc_target`, not sure how this maps on compiletest checks).

jieyouxu commented:
Changes since last reviewed:

- Removed `is_msvc` in favor of the more clear `is_windows_msvc`.
- Changed run-make tests to use `is_windows_msvc` only.

jieyouxu commented on src/tools/run-make-support/src/lib.rs:
```diff
@@ -3,9 +3,6 @@
-// We want to control use declaration ordering and spacing (and preserve use group comments), so
-// skip rustfmt on this file.
-#![cfg_attr(rustfmt, rustfmt::skip)]
 #![warn(unreachable_pub)]
```
The underlying reason is that rustfmt doesn't really tie comments to `use` statements, so they get moved all over the place.

Ignoring a whole file to rustfmt seems usually undesirable, so I think it's fine we don't have such magic comment.

jieyouxu replied:
I.e. https://github.com/rust-lang/rustfmt/issues/3720

bors commented:
:pushpin: Commit 3f4f23af617171e2e03e867a61b952d56166143b has been approved by `Kobzol`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Kobzol 3f4f23af617171e2e03e867a61b952d56166143b -->
<!-- homu: {"type":"Approved","sha":"3f4f23af617171e2e03e867a61b952d56166143b","approver":"Kobzol","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
